app/controllers/registrations_controller.rb

```diff
@@ -75,13 +76,14 @@ def check_allowed_email(resource)
     resource.errors.add(:email, "is not included in allowed domains.")
   end
 
-  def resolve_profile_field_issues
-    # Run this data update script when in a state of "first user" in the event
-    # that we are in a state where this was not already run.
-    # This is likely only temporarily needed.
-    return unless Settings::General.waiting_on_first_user
-
-    csv = Rails.root.join("lib/data/dev_profile_fields.csv")
-    ProfileFields::ImportFromCsv.call(csv)
+  def build_devise_resource
+    build_resource(sign_up_params)
+    resource.saw_onboarding = false
+    resource.registered_at = Time.current
+    resource.editor_version = "v2"
+    resource.remote_profile_image_url = Users::ProfileImageGenerator.call if resource.remote_profile_image_url.blank?
+    check_allowed_email(resource) if resource.email.present?
+    resource.save if resource.email.present?
+    yield resource if block_given?
```

Comment:
Are blocks given in the tests? If so they would be for the controller action `#create` and not the new method `#build_devise_resource`. I am probably oversimplifying things - but where would you call a controller action like create with a block (and now, since we've extracted to a method, I'm fairly sure we _don't_ ever call it with a block, since you are now the lone caller from `#create`, of this method).

This conditional makes a reader wonder where the blocks are getting passed from. I don't have any technical objection to leaving it in place but it felt misleading during code review.

Reply:
You raised a good question @djuber ! 

Turns out this code was initially taken directly from Devise. Based on this [SO answer](https://stackoverflow.com/questions/39425850/what-does-yield-resource-in-the-devise-controllers-do), it's for throwing hooks from a child class, a feature we definitely don't need. Removing!
```suggestion
```